src/Services/Testing/Testing.API/Controllers/TaskController.cs

```diff
@@ -0,0 +1,226 @@
+            return NotFound();
+        }
+    }
+
+    [HttpPost]
+    [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+    [ProducesResponseType(typeof(TaskResponse), (int)HttpStatusCode.OK)]
+    public async Task<ActionResult<TaskResponse>> CreateTaskAsync([FromBody] CreateTaskRequest request)
+    {
+        var title = Title.Create(request.Title);
+        if (title.IsFailure)
+        {
+            return BadRequest("Title is invalid");
+        }
+
+        var description = Description.Create(request.DescriptionText, request.DescriptionExamples,
+                                             request.DescriptionCases, request.DescriptionNote);
+        if (description.IsFailure)
+        {
+            return BadRequest("Description is invalid");
+        }
+
+        var difficultyData = await _dictionary.GetDifficultyByIdAsync(request.DifficultyId);
+        if (difficultyData == null)
+        {
+            return BadRequest("Difficulty is invalid");
+        }
+        var difficulty = Difficulty.Create(difficultyData.Id, difficultyData.Name);
+
+        var taskTypeData = await _dictionary.GetTaskTypeByIdAsync(request.TaskTypeId);
+        if (taskTypeData == null)
+        {
+            return BadRequest("TaskType is invalid");
+        }
+        var taskType = DomainType.Create(taskTypeData.Id, taskTypeData.Name);
+
+        var programmingLanguageData = await _dictionary.GetProgrammingLanguageByIdAsync(request.ProgrammingLanguageId);
+        if (programmingLanguageData == null)
+        {
+            return BadRequest("ProgrammingLanguage is invalid");
+        }
+        var programmingLanguage = ProgrammingLanguage.Create(programmingLanguageData.Id, programmingLanguageData.Name);
+
+        var solutionExample = SolutionExample.Create(request.SolutionExampleDescription, request.SolutionExample);
+        if (solutionExample.IsFailure)
+        {
+            return BadRequest("SolutionExample is invalid");
+        }
+
+        var executionCondition = ExecutionCondition.Create(request.ExecutionConditionTests, request.ExecutionTimeLimit);
+        if (executionCondition.IsFailure)
+        {
+            return BadRequest("ExecutionCondition is invalid");
+        }
+
+        var task = new DomainTask(
+            Guid.Empty,
+            title.Value!,
+            description.Value!,
+            difficulty.Value!,
+            taskType.Value!,
+            programmingLanguage.Value!,
+            solutionExample.Value!,
+            executionCondition.Value!
+            );
+
+        task = _tasks.Add(task);
+        await _tasks.UnitOfWork.SaveChangesAsync();
+
+        return Ok(task.Id);
+    }
+
+    [HttpPut]
+    [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+    [ProducesResponseType((int)HttpStatusCode.NotFound)]
+    [ProducesResponseType(typeof(TaskResponse), (int)HttpStatusCode.OK)]
+    public async Task<ActionResult<TaskResponse>> UpdateTaskAsync([FromBody] UpdateTaskRequest request)
```

Comment:
Common logic for update is
FintById
check for null
update properties
save changes